src/linker/Linker.Dataflow/DynamicallyAccessedMembersTypeHierarchy.cs

```diff
@@ -212,17 +208,16 @@ public DynamicallyAccessedMembersTypeHierarchy (LinkContext context, MarkStep ma
 			}
 
 			if (applied) {
-				using var _ = _scopeStack.PushScope (new MessageOrigin (type));
-				// Set up a context to report warnings on access to annotated members, with the annotated type as the origin.
-				var analysisContext = new ReflectionMethodBodyScanner.AnalysisContext (_scopeStack.CurrentScope.Origin, diagnosticsEnabled: true, _context);
-				ApplyDynamicallyAccessedMembersToType (ref reflectionMethodBodyScanner, analysisContext, type, annotation);
+				var origin = new MessageOrigin (type);
+				// Report warnings on access to annotated members, with the annotated type as the origin.
+				ApplyDynamicallyAccessedMembersToType (ref reflectionMethodBodyScanner, origin, type, annotation);
 				_typesInDynamicallyAccessedMembersHierarchy[type] = (annotation, true);
 			}
 
 			return applied;
 		}
 
-		void ApplyDynamicallyAccessedMembersToType (ref ReflectionMethodBodyScanner reflectionMethodBodyScanner, in ReflectionMethodBodyScanner.AnalysisContext analysisContext, TypeDefinition type, DynamicallyAccessedMemberTypes annotation)
+		void ApplyDynamicallyAccessedMembersToType (ref ReflectionMethodBodyScanner reflectionMethodBodyScanner, MessageOrigin origin, TypeDefinition type, DynamicallyAccessedMemberTypes annotation)
```

Comment:
nit: We were passing the analysisContext with the in parameter modifier, I think we still want that